JustSaying/Messaging/MessageHandling/IHandlerWithContext.cs

```diff
@@ -0,0 +1,25 @@
+using System.Threading.Tasks;
+
+namespace JustSaying.Messaging.MessageHandling
+{
+    public interface IHandlerWithContext<in T>
+    {
+        Task<bool> Handle(T message, MessageContext context);
+    }
+
+    public class HandlerAdapter<T> : IHandlerWithContext<T>
+    {
+        private readonly IHandlerAsync<T> _inner;
+
+        public HandlerAdapter(IHandlerAsync<T> inner)
+        {
+            _inner = inner;
+        }
+
+        public Task<bool> Handle(T message, MessageContext context)
```

Comment:
Referring back to David Fowler's recommendations, should this be `async` rather than just return the `Task`?